src/sensai/vector_model.py

```diff
@@ -125,6 +128,10 @@ def withTargetTransformer(self, targetTransformer: Optional[InvertibleDataFrameT
             the model, i.e. the transformation is completely transparent when applying the model.
         :return: self
         """
+        # Note: it is important to disallow targetTransformers for rule based models since we need
+        # predictedVarNames and modelOutputVarNames to coincide there.
```

Comment:
I don't understand this reasoning. The use of a transformer doesn't prevent them from being the same, and why do we need them to be the same?
In my mind, the reason why it is non-sensical to apply a target transformer is that since you are not training, you don't actually use any targets. And during inference, you obviously generate the targets yourself, so you could really only ever use the inverse (which an output transformer would be the better solution for).

Reply:
Apart from being nonsensical for rule based models (which I could have tolerated), it is actively damaging there. The modelOutputVars getter is used in VectorClassification, so it is crucial to get the names right. The predictedVars are provided by the user for rule-based models, not so the modelOutputVars.

The only difference between modelOutputVars and predictedVars is due to a possible pass through the target transformer. So I don't understand your statement - disallowing target transformers for rule-based model forces the two notions to agree and solves the problem described above

Reply:
The reasoning is illogical, because the use of a target transformer does not prevent them from being the same, like I said, and the only reason you need them to coincide is because you have modified `getModelOutputVariableNames` to force this (and perhaps just setting the member at construction - with an explanation that target transformers are disallowed and therefore they coincide - would be clearer).
The real reason why it should not be allowed to use target transformers the one I state in my second paragraph above.
Your reasoning in the comment we are talking about is due to a self-inflicted problem and therefore does not make sense without reading all the code.

Reply:
Long story short, I would propose 
* to change the comment to something like `# Disabled for rule-based models which do not apply fitting and therefore cannot make use of transformed targets`
* Explicitly set `_modelOuputVariableNames` at construction with the remark `# guaranteed to be the same as predictedVariableNames since target transformers and output transformers are disallowed`

Reply:
Right, I got what you meant now. Adjusted the comment and set the field at construction